src/Web/Types.hs

```diff
@@ -62,16 +64,26 @@ import           Data.Maybe          (listToMaybe)
 import           Data.Ord            (comparing)
 import qualified Data.Text           as T
 import           Data.Time.LocalTime
--- import qualified Data.Binary         as Bin
+import qualified Data.Binary         as Bin
+import           Data.Binary.Orphans()
+
+import           Control.DeepSeq
+import           Control.Monad
 -- import           Control.Monad
--- import           GHC.Generics (Generic)
+import           GHC.Generics (Generic)
 
 import           Text.Hamlet
 import           Yesod.Core
 import           Yesod.Static
 
 import           Theory
 
+
+-- | Derived Instances to fix things
+instance Bin.Binary ZonedTime where
+  get = liftM2 ZonedTime Bin.get Bin.get
+  put (ZonedTime d tod) = Bin.put d >> Bin.put tod
+
```

Comment:
This is a replacement for what was previously done in Instances.hs [0] to derive binary instances for the time types. Data.Binary.Orphans [1] provides most of them, but does not provide ZonedTime 

[0] https://github.com/tamarin-prover/tamarin-prover/pull/199/commits/ab6b4f38f4966497f252b0c7d1fee57584b8d573#diff-51df3a5cfa2326402ea8ce9eed3ebd64
[1] https://hackage.haskell.org/package/binary-orphans-0.1.6.0/docs/Data-Binary-Orphans.html